numpy/_core/src/common/gil_utils.c

```diff
@@ -0,0 +1,34 @@
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _MULTIARRAYMODULE
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <numpy/ndarraytypes.h>
+
+#include <stdarg.h>
+
+NPY_NO_EXPORT void
+npy_gil_error(PyObject *type, const char *format, ...)
+{
+    va_list va;
+    va_start(va, format);
```

Comment:
This looks good to me, thanks.

One thing though: I think at least technically every `va_start` call must be followed by a `va_end(va)` call before returning.

Reply:
Thanks!